public/app/features/dashboard/history/history.ts

```diff
@@ -0,0 +1,247 @@
+
+import './history_srv';
+
+import _ from 'lodash';
+import angular from 'angular';
+import moment from 'moment';
+
+import {DashboardModel} from '../model';
+import {HistoryListOpts, RevisionsModel} from './models';
+
+export class HistoryListCtrl {
+  appending: boolean;
+  dashboard: DashboardModel;
+  delta: { basic: string; html: string; };
+  diff: string;
+  limit: number;
+  loading: boolean;
+  max: number;
+  mode: string;
+  orderBy: string;
+  revisions: RevisionsModel[];
+  selected: number[];
+  start: number;
+
+  /** @ngInject */
+  constructor(private $scope,
+              private $rootScope,
+              private $window,
+              private $q,
+              private contextSrv,
+              private historySrv) {
+    $scope.ctrl = this;
+
+    this.appending = false;
+    this.diff = 'basic';
+    this.limit = 10;
+    this.loading = false;
+    this.max = 2;
+    this.mode = 'list';
+    this.orderBy = 'version';
+    this.selected = [];
+    this.start = 0;
+
+    this.resetFromSource();
+
+    $scope.$watch('ctrl.mode', newVal => {
+      $window.scrollTo(0, 0);
+      if (newVal === 'list') {
+        this.reset();
+      }
+    });
+
+    $rootScope.onAppEvent('dashboard-saved', this.onDashboardSaved.bind(this));
+  }
+
+  addToLog() {
+    this.start = this.start + this.limit;
+    this.getLog(true);
+  }
+
+  compareRevisionStateChanged(revision: any) {
+    if (revision.checked) {
+      this.selected.push(revision.version);
+    } else {
+      _.remove(this.selected, version => version === revision.version);
+    }
+    this.selected = _.sortBy(this.selected);
+  }
+
+  compareRevisionDisabled(checked: boolean) {
+    return (this.selected.length === this.max && !checked) || this.revisions.length === 1;
+  }
+
+  formatDate(date) {
+    date = moment.isMoment(date) ? date : moment(date);
+    const format = 'YYYY-MM-DD HH:mm:ss';
+
+    return this.dashboard.timezone === 'browser' ?
+      moment(date).format(format) :
+      moment.utc(date).format(format);
+  }
+
+  formatBasicDate(date) {
+    const now = this.dashboard.timezone === 'browser' ?  moment() : moment.utc();
+    const then = this.dashboard.timezone === 'browser' ?  moment(date) : moment.utc(date);
+    return then.from(now);
+  }
+
+  getDiff(diff: string) {
+    if (!this.isComparable()) { return; } // disable button but not tooltip
+
+    this.diff = diff;
+    this.mode = 'compare';
+    this.loading = true;
+
+    // instead of using lodash to find min/max we use the index
+    // due to the array being sorted in ascending order
+    const compare = {
+      new: this.selected[1],
+      original: this.selected[0],
+    };
+
+    if (this.delta[this.diff]) {
+      this.loading = false;
+      return this.$q.when(this.delta[this.diff]);
+    } else {
+      return this.historySrv.compareVersions(this.dashboard, compare, diff).then(response => {
+        this.delta[this.diff] = response;
+      }).catch(err => {
```

Comment:
error handlding that shows alerts like this are not needed, it's gone generically in the backendSrv